src/test/java/io/jenkins/plugins/analysis/warnings/recorder/MiscIssuesRecorderITest.java

```diff
@@ -235,6 +233,25 @@ public void shouldHaveOriginsIfBuildContainsWarnings() {
         assertThat(result).hasQualityGateStatus(QualityGateStatus.INACTIVE);
     }
 
+    /**
+     * This test is used to check if the build fails when the setFailOnError has been set to true.
+     */
+    @Test
+    public void shouldFailBuildWhenFailBuildOnErrorsIsSet() {
+        FreeStyleProject job = createFreeStyleProject();
+        enableEclipseWarnings(job);
+
```

Comment:
Let the job run here and verify that it is successful.
You can also check the `AnalysisResuls` for error messages so we see that this is the actual cause for the failure in line 252.